### PR TITLE
File Export: Add timestamps to ensure uniqueness

### LINKED ISSUE #6259: [Suggestion] - Be more permissive on file imports
###### Reproduction Steps

1. Export a file and have a name conflict with a file which already exists
2. File is named "collection.colpkg (1)" due to the SAF: https://stackoverflow.com/q/32849387. Ref: https://github.com/ankidroid/Anki-Android/blob/ac91fd6a51d6aa61c0eaa298aa9fcbf4aa9b30a4/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java#L1897-L1909
3. Import "collection.colpkg (1)"

###### Expected Result

4. Collection is imported

###### Actual Result

4. This isn't a valid Anki package file\n Filename "collection.colpkg (1)" doesn't have .apkg or .colpkg extension.

###### Debug info
AnkiDroid Version = 2.11alpha6

Android Version = 9

ACRA UUID = 0292ff8f-1cba-4ded-bf2b-b48989e552de

###### Research
- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
We shouldn't be more permissive with file imports, as this leaves an incompatibility between AnkiDroid and Anki imports, where AnkiDroid is more permissive of bad values.

## Fixes
Fixes #6259 (take 2)

## Approach

Revert the previous commit

Add a timestamp to the filename

## How Has This Been Tested?

On my device, tested all 3 imports:

* Full Collection with scheduling
* Full collection without scheduling
* Deck

## Learning (optional, can help others)
Think more about breaking compatibility when self-reviewing.

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by mikehardy** [comment]
Despite my comment on the related issue, I accept the revert for compatibility - playing nice in the ecosystem is more important to me than a weakly held philosophy

**Comment by david-allison-1** [comment]
> Despite my comment on the related issue, I accept the revert for compatibility - playing nice in the ecosystem is more important to me than a weakly held philosophy

Sorry! missed your comment and went on boldly

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/libanki/utils/TimeUtils.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5589182&bid=18314064)

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 1]
```diff
@@ -118,7 +121,9 @@
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.text.SimpleDateFormat;
 import java.util.ArrayList;
+import java.util.Date;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:126]: Codacy Issue found: [Avoid unused imports such as 'java.util.Date'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5589182)

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 2]
```diff
@@ -118,7 +121,9 @@
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.text.SimpleDateFormat;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:124]: Codacy Issue found: [Avoid unused imports such as 'java.text.SimpleDateFormat'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5589182)

**Comment by david-allison-1** [comment]
> > Despite my comment on the related issue, I accept the revert for compatibility - playing nice in the ecosystem is more important to me than a weakly held philosophy
> 
> Sorry! missed your comment and went on boldly

Your call on whether the revert commit is removed or not

**Comment by mikehardy** [comment]
No it's good - compatibility first unless it's important. This should make it unimportant, so -> onwards

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/libanki/utils/TimeUtils.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5589182&bid=18314064)

**Pull request merged**